File: C++/Program/Human.cpp
/* 
    Saya Hestina Dwi Hartiwi NIM 2108077 mengerjakan soal Latihan 2 dalam Praktikum
    mata kuliah Desain dan Pemrograman Berorientasi Objek untuk keberkahan-Nya maka 
    saya tidak melakukan kecurangan seperti yang telah dispesifikasikan. Aamin.
*/

/* import library */
#include <string>
#include <iostream>

using namespace std;

/*--- Deklarasi Class ---*/
class Human {
    /* Private Atributes */
    private:
        int NIK;
        string name, gender;
    public:
        /* Konstruktor */
        Human ()
        {
            this->NIK = 0;
            this->name = "";
            this->gender = "";
        }

        /* Getter and Setter */

        // Get Code
        int get_NIK()
        {
            return this->NIK;
        }

        // Get name
        string get_name()
        {
            return this->name;
        }

        // Get gender
        string get_gender()
        {
            return this->gender;
        }
        
        // Set Code
        void set_NIK(int NIK)
        {
            this->NIK = NIK;
        }

        // Set name
        void set_name(string name)
        {
            this->name = name;
        }

        // Set gender
        void set_gender(string gender)
        {
            this->gender = gender;
        }

        void printHuman()
        {
            cout << "NIK                : " << this->NIK << '\n';
            cout << "Gender             : " << this->gender << '\n';
        }

        /* Destructor */
        ~Human()
        {

        }
};

File: Python/Program/Human.py
# /* 
#     Saya Hestina Dwi Hartiwi NIM 2108077 mengerjakan soal Latihan 2 dalam Praktikum
#     mata kuliah Desain dan Pemrograman Berorientasi Objek untuk keberkahan-Nya maka 
#     saya tidak melakukan kecurangan seperti yang telah dispesifikasikan. Aamin.
# */

class Human:
    __NIK = 0
    __name = ""
    __gender = ""

    # Konstruktor
    def __init__(self):
        self.__NIK = 0
        self.__name = ""
        self.__gender = ""
    
    # Getter and Setter #

    # Get NIK
    def get_nik (self):
        return self.__NIK
    
    # Get Name
    def get_name (self):
        return self.__name

    # Get Gender
    def get_gender (self):
        return self.__gender
    
    # Set NIK
    def set_nik (self, NIK):
        self.__NIK = NIK

    # Set Name
    def set_name (self, name):
        self.__name = name
    
    # set Gender
    def set_gender (self, gender):
        self.__gender = gender
    
    # Public Method #
    def printName (self):
        print("Name             : ", self.__name)
    def printHuman(self):
        print("NIK              : ", self.get_nik())
        print("Gender           : ", self.get_gender())